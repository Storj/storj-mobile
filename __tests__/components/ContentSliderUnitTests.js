
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ContentSlider from '../../src/components/ContentSliderComponent';
import sinon from 'sinon';

Enzyme.configure({ adapter: new Adapter() });

describe('ContentSlider', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <ContentSlider 
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('buttons number is correct №1', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]} />
        );

        expect(wrapper.find('TouchableHighlight').length).toBe(2);
    });

    it('buttons number is correct №2', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]} />
        );

        expect(wrapper.find('TouchableHighlight').length).toBe(3);
    });

    it('buttons works correct', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]}
                position = { 0 }
                width = { 600 }
                height = { 600 } />
        );

        wrapper.instance().ref = { scrollTo: () => {} }

        let iterator = 0;
        wrapper.find('TouchableHighlight').forEach(child => {
            if(iterator === 1) {
                child.simulate('PressOut');
                expect(wrapper.instance().state.position).toBe(1);
            }
            
            iterator++;
        });      

        iterator = 0;
        
        wrapper.find('TouchableHighlight').forEach(child => {
            if(iterator === 0) {
                child.simulate('PressOut');
                expect(wrapper.instance().state.position).toBe(0);
            }
            
            iterator++;
        });      
    });

    it('clicking the same button doesnt change position №1', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]}
                position = { 1 }
                width = { 600 }
                height = { 600 } />
        );

        var instance = wrapper.instance();
        instance.ref = { scrollTo: () => {} }

        instance.onButtonClick(1);

        expect(wrapper.instance().state.position).toBe(1);
    });

    it('clicking the same button doesnt change position №1', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]}
                position = { 0 }
                width = { 600 }
                height = { 600 } />
        );

        var instance = wrapper.instance();
        instance.ref = { scrollTo: () => {} }

        instance.onButtonClick(0);

        expect(wrapper.instance().state.position).toBe(0);
    });

    it('swipe right works correct', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]}
                position = { 0 }
                width = { 600 }
                height = { 600 } />
        );

        var instance = wrapper.instance();
        instance.ref = { scrollTo: () => {} }

        const swipeSpy = sinon.spy(instance, 'onSwipe');
        instance.onSwipe({nativeEvent:{contentOffset:{x: 10}}});

        expect(wrapper.instance().state.position).toBe(1);
    });

    it('swipe left works correct', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]}
                position = { 1 }
                width = { 600 }
                height = { 600 } />
        );

        var instance = wrapper.instance();
        instance.ref = { scrollTo: () => {} }

        const swipeSpy = sinon.spy(instance, 'onSwipe');
        instance.onSwipe({nativeEvent:{contentOffset:{x: -10}}});

        expect(wrapper.instance().state.position).toBe(0);
    });

    it('swipe left doesnt work from first position', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]}
                position = { 0 }
                width = { 600 }
                height = { 600 } />
        );

        var instance = wrapper.instance();
        instance.ref = { scrollTo: () => {} }

        const swipeSpy = sinon.spy(instance, 'onSwipe');
        instance.onSwipe({nativeEvent:{contentOffset:{x: -10}}});

        expect(wrapper.instance().state.position).toBe(0);
    });

    it('swipe right doesnt work from last position', () => {
        const wrapper = shallow(
            <ContentSlider
                content = {[ 
                    () => { return(<View/>) },
                    () => { return(<View/>) }
                ]}
                position = { 1 }
                width = { 600 }
                height = { 600 } />
        );

        var instance = wrapper.instance();
        instance.ref = { scrollTo: () => {} }

        const swipeSpy = sinon.spy(instance, 'onSwipe');
        instance.onSwipe({nativeEvent:{contentOffset:{x: 10}}});

        expect(wrapper.instance().state.position).toBe(1);
    });
});