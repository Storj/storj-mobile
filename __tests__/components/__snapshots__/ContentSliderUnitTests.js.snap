// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentSlider renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentSlider
    content={
      Array [
        [Function],
        [Function],
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ScrollView
          decelerationRate={0.99}
          horizontal={true}
          onMomentumScrollEnd={[Function]}
          scrollEnabled={true}
          showsHorizontalScrollIndicator={false}
          style={Object {}}
        >
          <Component
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          >
            [Function]
          </Component>
          <Component
            style={
              Object {
                "height": undefined,
                "width": undefined,
              }
            }
          >
            [Function]
          </Component>
        </ScrollView>,
        <Component
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 74,
              "paddingVertical": 40,
            }
          }
        >
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPressOut={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#c6d5df",
                  "borderRadius": 4,
                  "borderWidth": 1.3,
                  "height": 16,
                  "marginRight": 22,
                  "width": 16,
                },
                Object {
                  "backgroundColor": "#c6d5df",
                  "opacity": 1,
                },
              ]
            }
            underlayColor="#ccc"
          >
            <Component />
          </TouchableHighlight>
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            onPressOut={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#c6d5df",
                  "borderRadius": 4,
                  "borderWidth": 1.3,
                  "height": 16,
                  "marginRight": 22,
                  "width": 16,
                },
                null,
              ]
            }
            underlayColor="#ccc"
          >
            <Component />
          </TouchableHighlight>
        </Component>,
      ],
      "style": Array [
        Object {
          "flex": 1,
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            >
              [Function]
            </Component>,
            <Component
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            >
              [Function]
            </Component>,
          ],
          "decelerationRate": 0.99,
          "horizontal": true,
          "onMomentumScrollEnd": [Function],
          "scrollEnabled": true,
          "showsHorizontalScrollIndicator": false,
          "style": Object {},
        },
        "ref": [Function],
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": [Function],
              "style": Object {
                "height": undefined,
                "width": undefined,
              },
            },
            "ref": null,
            "rendered": [Function],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": [Function],
              "style": Object {
                "height": undefined,
                "width": undefined,
              },
            },
            "ref": null,
            "rendered": [Function],
            "type": [Function],
          },
        ],
        "type": [MockFunction],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPressOut={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#c6d5df",
                    "borderRadius": 4,
                    "borderWidth": 1.3,
                    "height": 16,
                    "marginRight": 22,
                    "width": 16,
                  },
                  Object {
                    "backgroundColor": "#c6d5df",
                    "opacity": 1,
                  },
                ]
              }
              underlayColor="#ccc"
            >
              <Component />
            </TouchableHighlight>,
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPressOut={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#c6d5df",
                    "borderRadius": 4,
                    "borderWidth": 1.3,
                    "height": 16,
                    "marginRight": 22,
                    "width": 16,
                  },
                  null,
                ]
              }
              underlayColor="#ccc"
            >
              <Component />
            </TouchableHighlight>,
          ],
          "style": Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 74,
            "paddingVertical": 40,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.85,
              "children": <Component />,
              "delayPressOut": 100,
              "onPressOut": [Function],
              "style": Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#c6d5df",
                  "borderRadius": 4,
                  "borderWidth": 1.3,
                  "height": 16,
                  "marginRight": 22,
                  "width": 16,
                },
                Object {
                  "backgroundColor": "#c6d5df",
                  "opacity": 1,
                },
              ],
              "underlayColor": "#ccc",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.85,
              "children": <Component />,
              "delayPressOut": 100,
              "onPressOut": [Function],
              "style": Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#c6d5df",
                  "borderRadius": 4,
                  "borderWidth": 1.3,
                  "height": 16,
                  "marginRight": 22,
                  "width": 16,
                },
                null,
              ],
              "underlayColor": "#ccc",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ScrollView
            decelerationRate={0.99}
            horizontal={true}
            onMomentumScrollEnd={[Function]}
            scrollEnabled={true}
            showsHorizontalScrollIndicator={false}
            style={Object {}}
          >
            <Component
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            >
              [Function]
            </Component>
            <Component
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            >
              [Function]
            </Component>
          </ScrollView>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingHorizontal": 74,
                "paddingVertical": 40,
              }
            }
          >
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPressOut={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#c6d5df",
                    "borderRadius": 4,
                    "borderWidth": 1.3,
                    "height": 16,
                    "marginRight": 22,
                    "width": 16,
                  },
                  Object {
                    "backgroundColor": "#c6d5df",
                    "opacity": 1,
                  },
                ]
              }
              underlayColor="#ccc"
            >
              <Component />
            </TouchableHighlight>
            <TouchableHighlight
              activeOpacity={0.85}
              delayPressOut={100}
              onPressOut={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#c6d5df",
                    "borderRadius": 4,
                    "borderWidth": 1.3,
                    "height": 16,
                    "marginRight": 22,
                    "width": 16,
                  },
                  null,
                ]
              }
              underlayColor="#ccc"
            >
              <Component />
            </TouchableHighlight>
          </Component>,
        ],
        "style": Array [
          Object {
            "flex": 1,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              >
                [Function]
              </Component>,
              <Component
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              >
                [Function]
              </Component>,
            ],
            "decelerationRate": 0.99,
            "horizontal": true,
            "onMomentumScrollEnd": [Function],
            "scrollEnabled": true,
            "showsHorizontalScrollIndicator": false,
            "style": Object {},
          },
          "ref": [Function],
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": [Function],
                "style": Object {
                  "height": undefined,
                  "width": undefined,
                },
              },
              "ref": null,
              "rendered": [Function],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": [Function],
                "style": Object {
                  "height": undefined,
                  "width": undefined,
                },
              },
              "ref": null,
              "rendered": [Function],
              "type": [Function],
            },
          ],
          "type": [MockFunction],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPressOut={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#c6d5df",
                      "borderRadius": 4,
                      "borderWidth": 1.3,
                      "height": 16,
                      "marginRight": 22,
                      "width": 16,
                    },
                    Object {
                      "backgroundColor": "#c6d5df",
                      "opacity": 1,
                    },
                  ]
                }
                underlayColor="#ccc"
              >
                <Component />
              </TouchableHighlight>,
              <TouchableHighlight
                activeOpacity={0.85}
                delayPressOut={100}
                onPressOut={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#c6d5df",
                      "borderRadius": 4,
                      "borderWidth": 1.3,
                      "height": 16,
                      "marginRight": 22,
                      "width": 16,
                    },
                    null,
                  ]
                }
                underlayColor="#ccc"
              >
                <Component />
              </TouchableHighlight>,
            ],
            "style": Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 74,
              "paddingVertical": 40,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.85,
                "children": <Component />,
                "delayPressOut": 100,
                "onPressOut": [Function],
                "style": Array [
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#c6d5df",
                    "borderRadius": 4,
                    "borderWidth": 1.3,
                    "height": 16,
                    "marginRight": 22,
                    "width": 16,
                  },
                  Object {
                    "backgroundColor": "#c6d5df",
                    "opacity": 1,
                  },
                ],
                "underlayColor": "#ccc",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.85,
                "children": <Component />,
                "delayPressOut": 100,
                "onPressOut": [Function],
                "style": Array [
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "#c6d5df",
                    "borderRadius": 4,
                    "borderWidth": 1.3,
                    "height": 16,
                    "marginRight": 22,
                    "width": 16,
                  },
                  null,
                ],
                "underlayColor": "#ccc",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
